src/chainlit/frontend/src/components/organisms/chat/message/uploadButton.tsx

```diff
@@ -126,12 +143,12 @@ function _UploadButton({ askUser }: Props) {
   );
 }
 
-export default function UploadButton() {
+export default function UploadButton({ onlyButton }: { onlyButton?: boolean }) {
   const askUser = useRecoilValue(askUserState);
```

Comment:
Because we are reusing the state from the ask messages, this hides the spontaneous file upload button when we are waiting for an `AskUserMessage`. Example code:

```python
import chainlit as cl


@cl.on_file_upload(accept=["text/plain"], max_files=3)
async def upload_file(files: any):
    for file_data in files:
        file_name = file_data["name"]
        print(f"Uploaded file: {file_name}\nContent: {file_data['content']}\n\n")


@cl.on_chat_start
async def start():
    await cl.AskUserMessage(content="What is your name?", timeout=100).send()
```

My suggestion would be to have two different upload buttons instead.

Reply:
You're right, I've separated them now and added a `useUpload` hook, let me know what you think about this new version 🫡